0000-modules/motivation.md

```diff
@@ -0,0 +1,146 @@
+> submodules, but if it was only imported in a submodule it wouldn't work. 
+
+> I especially find the modules and crates design weird and verbose
+
+> fix the module system
+
+One user even states that the reason they stopped using Rust was that the
+"module system is really unintuitive." Our opinion is that if any user has
+decided to *stop using Rust* because they couldn't make modules work, the
+module system deserves serious reconsideration.
+
+### Found feedback
+
+@aturon devoted some time to searching for feedback from Rust users on various
+online forums about Rust (GitHub, the discourse forums, reddit, etc). He found
+numerous examples of users expressing confusion or frustration about how the
+module system works today. This data is collected in [this
+gist][learning-modules].
+
+## Underlying problems
+
+### Lots of syntax, all of it frontloaded
+
+The current Rust module system has four primary keywords: `extern crate`,
+`mod`, `use`, and `pub`. All of these are likely to be encountered very early
+by users - even if you're working on a small binary project, just to learn, you
+are likely going to want to add a dependency and break your code into two
+files. Doing those two things introduces the need to use  all four keywords
+that are a part of the module system, and to understand what each of them does.
+
+In our proposal, there are three primary keywords: `use`, `pub`, and `export`.
+And in an early project, only `use` and `pub` are going to be necessary. Not
+only that, the `export` keyword is just a visibility, like `pub`, and can be
+viewed as an extension of the pre-existing knowledge about visibility learned
+when your crate gains an external API.
+
+We believe we will simplify the system by requiring users to learn and do less
+stuff. 
+
+We also believe reducing the syntax improves the ergonomics for advanced users
+as well. Many of these declarations tend to feel like 'boilerplate' - you write
+the only thing you could possibly write. You may forget to write a `mod`
+statement, leading to a compiler error (worsening your edit-compile-debug
+cycle). We, the RFC authors, frequently have this experience.
+
+## Path confusion
+
+A problem, recognized by many users, is a confusion that exists about how paths
+work in Rust. Though this has often been glossed as confusion about the
+difference between absolute and relative paths, we don't believe this is true.
+Very many languages use absolute paths for their import statement, but relative
+paths internally to a module, without demonstrating the same confusion. We
+believe the confusion has two causes, both addressed by this RFC.
+
+### Differences between crate root and other modules
+
+In the crate root, absolute and relative paths are the same today. In other
+languages this tends not to be the case. Instead, a common system is to place
+the root of the current package under some namespace; this RFC proposes to do
+just that, using the new `crate` namespace.
```

Comment:
This is inconsistent with the rest of this RFC, which uses `local`. (Though, I still prefer `crate` :) )